.github/workflows/add-hip-number.yml

```diff
@@ -0,0 +1,97 @@
+name: Assign HIP Number and Rename File
+
+on:
+  pull_request:
+    types: [opened, synchronize]
+
+jobs:
+  assign-hip-number:
+    runs-on: ubuntu-latest
```

Comment:
should be on the self hosted runner

`runs-on: [self-hosted, Linux, medium, ephemeral`